modules/tinymce/src/themes/silver/main/ts/ui/core/ControlUtils.ts

```diff
@@ -22,9 +22,29 @@ const onSetupFormatToggle = (editor: Editor, name: string) => (api: Toolbar.Tool
   };
 
   // The editor may or may not have been setup yet, so check for that
-  editor.initialized ? init() : editor.on('init', init);
+  editor.initialized ? init() : editor.once('init', init);
 
-  return boundCallback.clear;
+  return () => {
+    editor.off('init', init);
```

Comment:
This was actually a bug, as due to refreshes happening the button would have been setup and destroyed at least once before init. This meant that the `init` code actually ran twice since it wasn't unbound.

Reply:
Is this still necessary if you changed it above to use `editor.once`?

Reply:
Yes, as once only removes it when it's actually called. If you read the above, I mentioned it's binding but not ever running. As such even using once it would still bind multiple times if the toolbar button is setup and destroyed before the `init` event is fired.

Reply:
I should add the reason I changed it to `once` is so that we don't keep a reference for the entire life of the editor once it does actually run.